# <center>Twin brothers and bayes theorem</center>

Suppose we are asked the question: <b>Elvis Presley had a twin brother who died at birth.  What is the probability that Elvis was an identical twin?</b>

In order to make our problem easier, we will clarify a few facts about identical twins. Identical twins are known as monozygotic twins, meaning that they both devolop from a single zygote. As a result, monozygotic twins are the same gender, either male-male or female-female. So, we rephrase our question: <b>What percentage of male-male twins are monozygotic.</b>

In addition, here is an important fact: .08% of twins are monozygotic.

## <center>Without Bayes' theorem</center>

We use a tree to visualize the problem.

<img src="treeReport1.jpg" alt="Probability Tree" height="400" width="400">

Assuming we have 100 twins, lets calculate the number of male-male dizygotic, male-male monozygotic, and total number of male-male twins. 

In [1]:
# calculate number of male-male dizygotic twins using the percentage of dizygotic and percentage of male-male
DiMM = 100 * .92 * .25

# calculate number of male-male monozygotic twins using the percentage of monozygotic and percentage of male-male
MoMM = 100 * .08 * .5

# calculate total number of male-male twins
TotalMM = DiMM + MoMM

print("Number of male-male dizygotic twins: {}".format(DiMM))
print("Number of male-male monozygotic twins: {}".format(MoMM))
print("Total number of male-male twins: {}".format(TotalMM))

Number of male-male dizygotic twins: 23.0
Number of male-male monozygotic twins: 4.0
Total number of male-male twins: 27.0


In [2]:
# next we can calculate the fraction of male-male twins that are monozygotic
fractionMoMM = MoMM / TotalMM
percentMoMM = fractionMoMM * 100
print("Percentage of male-male monozygotic twins: {0:.1f}%".format(percentMoMM))

Percentage of male-male monozygotic twins: 14.8%


So, we can conclude that Elvis had a 14.8% chance to identical twins with his brother.

## <center>With Bayes' theorem</center>
However, rather than using a huge tree, we can use Bayes' theorem to make a much more eligant solution. First, assuming we are only dealing with twins, we find must find P(male-male|monozygotic), P(male-male), and P(monozygotic).  Then we can calculate P(monozygotic|male-male).

In [6]:
from thinkbayes2 import Pmf

% matplotlib inline

In [7]:
twins = Pmf()

# first calculate the total percentage of male-male twins. We can do this by adding the percentage of male-male
# monozygotic and the percentage of male-male dizygotic
twins.Set('male-male', .08*.50 + .92*.25)
twins.Set('male-male|monozygotic', .50)
twins.Set('monozygotic', .08)

print(twins['male-male'])
print(twins['male-male|monozygotic'])
print(twins['monozygotic'])

0.27
0.5
0.08


In [8]:
# now using bayes theorem
temp = twins['male-male|monozygotic'] * twins['monozygotic'] / twins['male-male']
print("P(monozygotic|male-male): {0:.3f}".format(temp))

P(monozygotic|male-male): 0.148


# <center>The M&M problem

First encode color mixes as dictionaries

In [ ]:
mix94 = dict(brown=30,
             yellow = 20,
             red=20;)